Confirm delete image

GDSNewt opened the pull request:
Trello Card: https://trello.com/c/duxwb607/1443-confirm-before-deleting-an-image

When interacting with images via the markdown editor the option to "Delete image" will now route the user to a confirmation page. Previously clicking this link would have deleted the image and displayed a flash message in the modal. Adding a confirmation page for deletion is in keeping with changes made to other delete actions, such as deleting drafts and deleting attachments. 

<img width="981" alt="Screenshot 2020-03-30 at 14 52 35" src="https://user-images.githubusercontent.com/41922771/78007385-24bcc280-7336-11ea-9324-069fc25127b5.png">

The user will also be routed to this confirmation page in a non-modal context when deleting a lead image.

benthorner commented:
Just a passing comment from me: could we add a link to the Trello card in the description of this PR? We do this so that we have a link back to 'source of the work', for future reference.

(It also means the PR gets listed on the card, automagically.)

GDSNewt commented on app/assets/javascripts/modules/inline-image-modal.js:
```diff
@@ -35,16 +35,16 @@ InlineImageModal.prototype.actionCallback = function (item) {
     'delete': function () {
+      this.workflow.render(window.ModalFetch.getLink(item))
+    },
+    'confirm-delete': function () {
```
ha, never heard of kebab case before 😄. Will amend them both to use camelCase

ChrisBAshton commented on app/views/images/_image.html.erb:
```diff
@@ -27,19 +27,9 @@
-    <button class="govuk-link app-link--button app-link--destructive">Delete image</button>
-  <% end %>
-<% end %>
+<% actions << link_to("Delete image", confirm_delete_image_path(edition.document, image.image_id),
```
I see this suggestion has been resolved, but the code doesn't seem to have changed?

GDSNewt replied:
I have left the code as it is as it is more in keeping with the existing format seen elsewhere.

ChrisBAshton commented on app/views/file_attachments/confirm_delete.html.erb:
```diff
@@ -14,7 +14,7 @@
       multipart: true,  
       data: {
-        "modal-action": "confirm-delete",
+        "modal-action": "confirmDelete",
```
Again I would split this out into a separate, tidy-up commit, rather than bundle it in with the "Add delete confirmation functionality". Not essential but it is a good habit to get into.

GDSNewt replied:
Thanks Chris, having discussed this with Ben I think I will leave it this time but will definitely start doing it in my future PRs 👍

GDSNewt commented:
> Thanks for addressing the comments 👍 a couple of the conversations have been 'resolved' without the code being changed, was that deliberate? (Fair enough if you disagree with them, just worth adding a response to the comments to clarify 😊)

Ah yes, I suppose that would be helpful. I judged that I should leave the code as is, just because the format I used was the one already used in the existing code. I was hesitant to start change the format of all the actions and so chose to be conservative in this regard.

GDSNewt commented on app/assets/javascripts/modules/inline-image-modal.js:
```diff
@@ -35,16 +35,16 @@ InlineImageModal.prototype.actionCallback = function (item) {
-    'cropBack': function () {
-      this.workflow.render(window.ModalFetch.getLink(item))
-    },
-    'metaBack': function () {
```
This will be something I do moving forward 👍

GDSNewt commented on config/routes.rb:
```diff
@@ -68,6 +68,8 @@
     get "/images/:image_id/edit" => "images#edit", as: :edit_image
     patch "/images/:image_id/edit" => "images#update"
     delete "/images/:image_id" => "images#destroy", as: :destroy_image
+    get "/images/:image_id/delete" => "images#confirm_delete", as: :confirm_delete_image
```
Thanks for pointing this out, I have now implemented a request test

GDSNewt commented on app/views/images/_lead_image.html.erb:
```diff
@@ -7,12 +7,12 @@
                       data: { gtm: "download-lead-image" }) %>
 
-<% actions << link_to("Delete lead image", confirm_delete_image_path(@edition.document, lead_image.image_id),
+<% unless rendering_context == "modal" %>
```
I used fixup on this commit to squash it into the previous commit 👍

benthorner commented on app/views/images/_lead_image.html.erb:
```diff
@@ -9,6 +9,10 @@
 <% unless rendering_context == "modal" %>
 
+  <% actions << link_to("Delete lead image", confirm_delete_image_path(@edition.document, lead_image.image_id),
+                      class: "govuk-link app-link--button app-link--destructive",
```
Alignment issue here.
